Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/QuickPulseProcessProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib;
+
+    /// <summary>
+    /// Top CPU collector.
+    /// </summary>
+    internal sealed class QuickPulseProcessProvider : IQuickPulseProcessProvider
+    {
+        private readonly IQuickPulsePerfLib perfLib;
+
+        public QuickPulseProcessProvider(IQuickPulsePerfLib perfLib)
+        {
+            this.perfLib = perfLib ?? PerfLib.PerfLib.GetPerfLib();
+        }
+
+        public void Initialize()
+        {
+            this.perfLib.Initialize();
```

Comment:
Reviewed again - and turns out the delayed initialization is in fact needed. We only need to initialize if collection actually occurred. Some (possibly most) applications will go all their lives without ever collecting, so we don't want to initialize anything in that case; in fact, we don't even start the collection thread. @SergeyKanzhelev was super glad about that way back when :)